pages/constructor-encoder.page.tsx

```diff
@@ -0,0 +1,171 @@
+import { Interface } from '@ethersproject/abi';
+import { BigNumber } from '@ethersproject/bignumber';
+import { isValidAddress, stripHexPrefix } from 'ethereumjs-util';
+import _ from 'lodash';
+import { ChangeEvent, useState } from 'react';
+
+import DecoderSvg from '../public/static/svg/decoders';
+import { Button } from '../src/components/lib/Button';
+import { Input } from '../src/components/lib/Input';
+import { ToolLayout } from '../src/layout/ToolLayout';
+import { encodeConstructor } from '../src/lib/encodeContructor';
+import { parseAbi } from '../src/lib/parseAbi';
+
+export interface ConstructorEncoderProps {}
+
+export default function ConstructorEncoder() {
+  const [error, setError] = useState<string | undefined>();
+  const [encodedResult, setEncodedResult] = useState<string[] | undefined>();
+
+  const [rawAbi, setRawAbi] = useState<string>();
+  const [iface, setIface] = useState<Interface>();
+  const [errors, setErrors] = useState<string[]>([]);
+  const [values, setValues] = useState<string[]>([]);
+
+  async function handleEncodeConstructor() {
+    if (!rawAbi) {
+      setError('Please provide ABI');
+      return;
+    }
+    if (_.some(values, _.isUndefined)) {
+      setError('Some values are empty');
+      return;
+    }
+    const iface = parseAbi(rawAbi) as Interface;
+    try {
+      const encoded = encodeConstructor(iface, values as string[]);
+      setEncodedResult(stripHexPrefix(encoded).match(/.{1,64}/g) as string[]);
+    } catch (e) {
+      setError(`${(e as any).code} ${(e as any).argument}`);
+    }
+  }
+
+  const encodeButtonDisabled = !rawAbi || _.some(values, _.isUndefined);
```

Comment:
There's another `_.some` here, so we should repeat the treatment we estabilish when talking about line 30.